examples/neural_compressor/text-classification/intent-classification/run_clinc.py

```diff
@@ -0,0 +1,674 @@
+        teacher_tokenizer = AutoTokenizer.from_pretrained(
+            optim_args.teacher_model_name_or_path,
+            use_fast=model_args.use_fast_tokenizer,
+        )
+        teacher_model = AutoModel.from_pretrained(
+            optim_args.teacher_model_name_or_path,
+            from_tf=bool(".ckpt" in model_args.model_name_or_path),
+            config=teacher_config,
+        )
+        teacher_model = SetFitModelTraining(teacher_model, teacher_tokenizer)
+
+        teacher_model.to(training_args.device)
+
+        distillation_config = IncDistillationConfig.from_pretrained(
+            optim_args.distillation_config if optim_args.distillation_config is not None else default_config,
+            config_file_name="distillation.yml",
+            cache_dir=model_args.cache_dir,
+        )
+
+        examples = raw_datasets["train"]["text"]
+        examples_duplicate_ratio = 100
+        examples_duplicate = []
+        for i in range(int(examples_duplicate_ratio)):
+            examples_duplicate.extend(examples)
+        examples_duplicate.extend(
+            examples[: int(len(examples) * (examples_duplicate_ratio - int(examples_duplicate_ratio)))]
+        )
+        if data_args.max_train_samples is not None:
+            examples_duplicate = raw_datasets["train"].select(range(data_args.max_train_samples))["text"]
+        shuffled_examples_duplicate = copy.deepcopy(examples_duplicate)
+        distillation_dataset = list(zip(examples_duplicate, shuffled_examples_duplicate))
+
+        def sentences_data_collator(sentences_pairs):
+            return {"sentences": [[sp[i] for sp in sentences_pairs] for i in range(len(sentences_pairs[0]))]}
+
+        def train_func(model):
+            model = SetFitModelTraining(model, tokenizer)
+            trainer.model_wrapped = model
+            trainer.model = model
+            trainer.train_dataset = distillation_dataset
+            data_collator = trainer.data_collator
+            trainer.data_collator = sentences_data_collator
+            checkpoint = None
+            if resume_from_checkpoint is not None:
+                checkpoint = resume_from_checkpoint
+            elif last_checkpoint is not None:
+                checkpoint = last_checkpoint
+            train_result = trainer.train(distiller.distillation, resume_from_checkpoint=checkpoint)
+            metrics = train_result.metrics
+            trainer.save_model()  # Saves the tokenizer too for easy upload
+            trainer.log_metrics("train", metrics)
+            trainer.save_metrics("train", metrics)
+            trainer.save_state()
+            trainer.data_collator = data_collator
+            trainer.train_dataset = train_dataset
+            return trainer.model.model.model
+
+        # Creation Distillation object used for IncTrainer training loop
+        distiller = IncDistiller(
+            teacher_model=teacher_model, config=distillation_config, eval_func=lambda x: 1, train_func=train_func
```

Comment:
Just to save some training time, also I didn't come up with a good way to evaluate the SetFit model, since it output the embeddings only, use the metric of the classifier on the downstream task would be misleading.